### ISSUE TITLE
unnecessary_safety_comment wrongly complains about safety comment in doctest

### ISSUE THREAD

**Issue by sandersaares** [rust-lang/rust-clippy#16553]
### Summary

I have a doctest that has a safety comment (on an unsafe block in the doctest) and Clippy complains it is not necessary on the code the doctest is documenting.

```toml
[lints.clippy]
unnecessary_safety_comment = "warn"
```

### Lint Name

unnecessary_safety_comment

### Reproducer

I tried this code:

```rust
//! ```
//! // SAFETY: All is well.
//! unsafe {
//!    foo()
//! }
//! ```

mod blah {}
```

I saw this happen:

```
C:\Source\Oss\random\asdf [master +4 ~0 -0 !]> cargo clippy
    Checking asdf v0.1.0 (C:\Source\Oss\random\asdf)
warning: module has unnecessary safety comment
 --> src\lib.rs:8:1
  |
8 | mod blah {}
  | ^^^^^^^^^^^
  |
help: consider removing the safety comment
 --> src\lib.rs:2:8
  |
2 | //! // SAFETY: All is well.
  |        ^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.93.0/index.html#unnecessary_safety_comment
  = note: requested on the command line with `-W clippy::unnecessary-safety-comment`

warning: `asdf` (lib) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.26s
```

I expected no warnings.

### Version

```text
rustc 1.93.0 (254b59607 2026-01-19)
binary: rustc
commit-hash: 254b59607d4417e9dffbc307138ae5c86280fe4c
commit-date: 2026-01-19
host: x86_64-pc-windows-msvc
release: 1.93.0
LLVM version: 21.1.8
```
